packages/web3-eth-abi/src/index.js

```diff
@@ -99,74 +99,88 @@ ABICoder.prototype.encodeParameter = function (type, param) {
  * @return {String} encoded list of params
  */
 ABICoder.prototype.encodeParameters = function (types, params) {
-    const bytesToEvenLength = (param) => {
-        if (Buffer.isBuffer(param)) {
-            param = utils.toHex(param)
+    const paramTypeBytes = new RegExp(/^bytes([0-9]*)$/);
+    const paramTypeBytesArray = new RegExp(/^bytes([0-9]*)\[\]$/);
+    const paramTypeNumber = new RegExp(/^(u?int)([0-9]*)$/);
+    const paramTypeNumberArray = new RegExp(/^(u?int)([0-9]*)\[\]$/);
+
+    params = params.map(function (param, index) {
+        let type = types[index]
+        if (typeof type === 'object' && type.type) {
+            // We may get a named type of shape {name, type}
+            type = type.type
         }
-        // bitwise AND operator returns true if odd
-        if (param.length & 1) { 
-            return '0x0' + param.substring(2)
-        }
-        return param
-    }
-    const bytes32ToFixedLength = (param) => {
-        if (Buffer.isBuffer(param)) {
-            param = utils.toHex(param)
-        }
-        if (param.length < 66) { // 66 includes `0x`
-            return utils.rightPad(param, 64)
+
+        // Format BN to string
+        if (utils.isBN(param) || utils.isBigNumber(param)) {
+            return param.toString(10);
         }
-        return param
-    }
 
-    return ethersAbiCoder.encode(
-        this.mapTypes(types),
-        params.map(function (param, index) {
-            let type = types[index]
-            if (typeof type === 'object' && type.type) {
-                // We may get a named type of shape {name, type}
-                type = type.type
+        // Handle some formatting of params as to not break previous compatability
```

Comment:
```suggestion
        // Handle some formatting of params for backwards compatability with Ethers V4
```

(Just trying to launch a GH Actions run here...it broke last Friday right as this PR was submitted)